spec/features/self_service/advisers_index_spec.rb

```diff
@@ -10,6 +10,36 @@
     then_i_can_see_the_advisers_for_each_trading_name
   end
 
+  scenario 'The principal can delete advisers' do
+    given_i_am_a_fully_registered_principal_user
+    and_i_have_a_firm_with_trading_names_and_advisers
+    and_i_am_logged_in
+    and_i_am_on_the_advisers_page
+    when_i_delete_the_first_adviser
+    then_i_am_on_the_advisers_page
+    and_i_can_see_a_success_message
+    and_i_cannot_see_the_deleted_adviser
+  end
+
+  def when_i_delete_the_first_adviser
+    @adviser_name = advisers_index_page.parent_firm.advisers.first.name.text
+    advisers_index_page.parent_firm.advisers.first.delete_link.click
+  end
+
+  def then_i_am_on_the_advisers_page
+    expect(advisers_index_page).to be_displayed
+  end
+
+  def and_i_can_see_a_success_message
+    expected_message = @adviser_name + ' has been successfully removed.'
```

Comment:
Yea — I can see arguments for and against both for this one, but not much reason to get into it since we're not localising.

Not relevant but interesting: I just found `I18n.t!` that raises an exception when the translation is missing, which can be useful in tests.